src/jit/compiler.cpp

```diff
@@ -2319,85 +2319,129 @@ const char* Compiler::compLocalVarName(unsigned varNum, unsigned offs)
 /*****************************************************************************/
 
 #ifdef _TARGET_XARCH_
-static bool configEnableISA(InstructionSet isa)
+static bool configEnableIsa(const JitFlags& jitFlags, InstructionSet isa)
 {
+    //
+    // NOTE: This function needs to be kept in sync with EEJitManager::SetCpuInfo() in vm\codemap.cpp
+    // NOTE: This function needs to be kept in sync with compSetProcesor() in jit\compiler.cpp
+    //
+
     switch (isa)
     {
-        case InstructionSet_AVX2:
```

Comment:
Previously this was checking the individual ISA and then falling through and checking each downlevel ISA. However, in `compSetProcessor` we didn't exactly follow this logic which was leading to some errors.

Now, this just checks the individual config flag and asserts that the downlevel ones are also correct, which puts the onus on the caller to validate that they are following the correct hierarchy.

Reply:
With your changes, `compSetProcessor` is now simply making things more complex rather than simplifying. It appears to me that we should eliminate this method, and simply fold what it's doing into `compSetProcessor`

Reply:
> With your changes, `compSetProcessor` is now simply making things more complex rather than simplifying

Could you clarify what you mean here?

This change removed half the checks from `compSetProcessor` and made them part of this method (`configEnableISA`) instead (and reduces the overall work that was being done in `Release` builds, while keeping the amount of work being done in `Debug` builds the same). `compSetProcessor` is simpler now, it just has a deeper if-block nesting as it is appropriately keeping the ISA hierarchy now.

Reply:
What I mean is that we are calling `configEnableIsa` from a place where we know what case we're in, and then we switch on that case - it just seems more convoluted than directly doing the check and assert in `compSetProcessor`, though IMO the asserts would then be unnecessary due to the logic of the control flow.

Reply:
Ah, ok. I can get rid of `configEnableIsa` then, and just have the checks be explicit.

Reply:
My original thinking was that we should have the logic centralized, so it can be reused... However, this happens very early and everyone else can just rely on the `compSupports()` checks (as they already do).

Reply:
Resolved.

Reply:
Excellent - thanks!